### PR TITLE
Add site examples

### PR DESCRIPTION
**Pull request by chrisprice**
Inspired by #991

### CODE REVIEW

File: site/src/examples/area-line-point/index.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+</style>
+
+{{{area-line-point-html}}}
+
+<script>
+{{{area-line-point-js}}}
+</script>
+
+In this example we combine a {{{ hyperlink 'series/area.html' title='area series' }}}, {{{ hyperlink 'series/line.html' title='line series' }}} and {{{ hyperlink 'series/point.html' title='point series' }}} using a {{{ hyperlink 'multi.html' title='multi series' }}}. This combination allows for flexibility when styling the series to achieve e.g. gradient fills. It also demonstrates {{{ hyperlink 'decorate-pattern.html' title='decorating' }}} the point series to easily add labels to specific points on the series.
+
```
Comment (ColinEberhardt) [inline site/src/examples/area-line-point/index.md]: I think one other very powerful connect it demonstrates which is worth mentioning is the multi-series `mapping` concept. In most examples we just use it to simple wire-up each series. However, in this case you are using it as a way to create a projection, which is more like a traditional `map`.

**Comment by ColinEberhardt** [comment]
Nice examples 👍 

One minor thing, rather than using:

```
var yExtent = fc.util.extent()
     .fields([
         function(d) { return d.y; }
     ])
     .pad(0.2);
```

You could do the following:

```
var yExtent = fc.util.extent()
     .fields(['y'])
     .pad(0.2);
```

Although I am pretty sure you know that already, and are using the functional form because you don't like the more 'magic' version!

File: site/src/examples/area-line-point/index.md (hunk 1 above)
Comment (chrisprice) [inline site/src/examples/area-line-point/index.md]: Yea good point. I wanted to use annotated snippets of the js code but couldn't remember if that was possible?

**Comment by chrisprice** [comment]
I'd really like to split out extent and eliminate a few of its "features" -
* Remove support for field names
* Remove support for nested arrays
* Split out date functionality e.g. import {linear, date} from 'd3fc-extent'. 
* Fix pad as being a tuple.
One for later in the week.

File: site/src/examples/area-line-point/index.md (hunk 1 above)
Comment (ColinEberhardt) [inline site/src/examples/area-line-point/index.md]: The `code block` helper outputs the contents of a file between `\\START` and `\\END` comments:

https://github.com/ScottLogic/d3fc/blob/master/site/build/handlebars-helpers/codeblock.js

It's a bit cumbersome, but does the trick.

**Comment by ColinEberhardt** [comment]
Updated looks good - 👍 to merging

**Pull request merged**